docs/cicd.md

```diff
@@ -0,0 +1,55 @@
+# CI/CD pipelines
+The capabilities of FFC are underpinned by DevOps CI/CD pipelines providing cloud agnostic tooling with build and release automation  through Jenkins.
+
+## Standards
+- Pipelines should be scripted within a Jenkinsfile and source controlled with the service it relates to
+- The Groovy syntax should be used
+- Build steps should be containerised where possible to reduce the plugin and Jenkins configuration dependencies
+
+## Using Jenkins locally
+### Installation
+It is recommended that Jenkins is run locally within the Blue Ocean Docker image.
+
+The below command will pull the image and run the container.
+
+```
+docker run \
+  -u root \
+  --rm \
+  -d \
+  -p 8080:8080 \
+  -p 50000:50000 \
+  -v jenkins-data:/var/jenkins_home \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -v <PATH_TO_LOCAL_GIT_REPOS>:repos:ro \
+  jenkinsci/blueocean
+```
+
+`-v <PATH_TO_LOCAL_GIT_REPOS>:repos:ro`  
+
+The above line should be amended to specify the file path of your local git repositories.  This will allow you to run builds from your local repository.
+
+An example could be ` -v /c/users/ddts_220606/source/repos:\repos:ro`  
+
+This [official guide](https://jenkins.io/doc/book/installing/) includes instructions for how to setup access credentials to the Jenkins container.
+
```

Comment:
Should we mention anything about plugins here? I've grabbed the password out of the logs, as instructed by Jenkins, then am presented with a choice of installing default or choosing my own plugins. Should we have a recommended list or just use default?

Do we intend to standardise across FFC or might different workstreams (or even projects) use different plugins? If they vary, I think we should at least instruct on where to find a list of required plugins (project Readme? workstream developer guide?)

I'm guessing it's a missing plugin that's causing me to get this error, running the deployment pipeline Matthew's added to the User Service:
```
java.lang.NoSuchMethodError: No such DSL method 'withKubeConfig' found ...
```

Reply:
We definitely should  include something about plugins, however as we haven't yet agreed what should be in our pipelines and which plugins they should use, I don't think we need to at this point.  

Currently the CI/CD section of the guide is only this one page, but I expect as we learn and make decisions it will expand to cover the full pipeline.

I'll add a ticket specifically for plugins.

Reply:
We definitely should  include something about plugins, however as we haven't yet agreed what should be in our pipelines and which plugins they should use, I don't think we need to at this point.  

Currently the CI/CD section of the guide is only this one page, but I expect as we learn and make decisions it will expand to cover the full pipeline.